win32ss/user/ntuser/caret.c

```diff
@@ -351,14 +351,7 @@ NtUserCreateCaret(
    }
    else
    {
-      if (nWidth == 0)
-      {
-          nWidth = UserGetSystemMetrics(SM_CXBORDER);
-      }
-      if (nHeight == 0)
-      {
-          nHeight = UserGetSystemMetrics(SM_CYBORDER);
-      }
+      UserSystemParametersInfo(SPI_GETCARETWIDTH, 0, &nWidth, 0);
```

Comment:
As previously mentioned, Windows doesn't appear to do that (and original code was compatible with Windows behaviour).